include/unifex/scheduler_concepts.hpp

```diff
@@ -199,6 +205,34 @@ UNIFEX_CONCEPT //
     UNIFEX_FRAGMENT(unifex::_scheduler_provider, SP);
 #endif
 
+namespace _schedule
+{
+  struct _fn {
+  private:
+    template <typename Scheduler>
+    static auto impl_(Scheduler sched)
+        noexcept(noexcept(make_sender_for<schedule>(
```

Comment:
> Does the use of make_sender_for here mean that the return-type of a customised schedule() call is no longer the same as the return-type of the customisation? 

Yes, and it makes me a little uncomfortable.

> If so could this wrapping inhibit subsequent customisations where they may have customised some other algorithm on that specific type?

True, it could.

> An alternative to using this helper to implement the get_scheduler() query would be requiring that all of the scheduler implementations return senders that have get_scheduler() customised.

Lots of senders could have `get_scheduler()` customized for lots of different reasons. We need an unambiguous way to identify senders returned from `schedule` do we can make the appropriate magic happen for `co_await schedule(sched)`.

One thought I've had is to give a sender a way to opt in to "schedule-sender-ness" intrusively instead of always relying on a wrapper. And yes, it would help to eliminate the redundant storage in most schedule senders.